js/coinsbit.js

```diff
@@ -0,0 +1,1099 @@
+        //           role: '2',
+        //           deal: '7.86328543'
+        //         },
+        //         ...
+        //       ]
+        const symbol = this.safeValue (market, 'symbol');
+        const id = this.safeString2 (trade, 'id', 'tid');
+        const orderId = this.safeString (trade, 'dealOrderId');
+        const timestamp = this.safeTimestamp2 (trade, 'time', 'date');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
+        const side = this.safeString (trade, 'type');
+        const cost = this.safeString (trade, 'deal');
+        const fee = this.safeString (trade, 'fee');
+        let takerOrMaker = '';
+        if ('role' in trade) {
+            takerOrMaker = this.safeInteger (trade, 'role') === 2 ? 'taker' : 'maker';
+        } else {
+            takerOrMaker = 'taker';
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': orderId,
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': fee,
+        }, market);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        if (type !== 'limit') {
+            throw new InvalidOrder (this.id + ' createOrder supports limit orders only');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+            'side': side,
+            'amount': this.amountToPrecision (symbol, amount),
+            'price': this.priceToPrecision (symbol, price),
+        };
+        const response = await this.privatePostOrderNew (this.extend (request, params));
+        // {
+        //     success: true,
+        //     message: '',
+        //     result: {
+        //       orderId: '8629354782',
+        //       market: 'ETH_USDT',
+        //       price: '4000',
+        //       side: 'buy',
+        //       type: 'limit',
+        //       timestamp: '1640212275.928',
+        //       dealMoney: '7.98856357',
+        //       dealStock: '0.002',
+        //       amount: '0.002',
+        //       takerFee: '0.002',
+        //       makerFee: '0.002',
+        //       left: '0',
+        //       dealFee: '0'
+        //     },
+        //     code: '200'
+        //   }
+        const data = this.safeValue (response, 'result', {});
+        return this.parseOrder (data, market);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new InvalidOrder (this.id + ' createOrder supports limit orders only');
```

Comment:
This line looks copypasted from createOrder, but the message and the exception type should be different:
```JavaScript
if (symbol === undefined) {
    throw new ArgumentsRequired (this.id + ' cancelOrder() requires a symbol argument');
}
```
```